packages/google_maps_flutter/google_maps_flutter/example/lib/heatmap.dart

```diff
@@ -0,0 +1,163 @@
+    const WeightedLatLng(LatLng(37.782, -122.435)),
+    const WeightedLatLng(LatLng(37.785, -122.447)),
+    const WeightedLatLng(LatLng(37.785, -122.445)),
+    const WeightedLatLng(LatLng(37.785, -122.443)),
+    const WeightedLatLng(LatLng(37.785, -122.441)),
+    const WeightedLatLng(LatLng(37.785, -122.439)),
+    const WeightedLatLng(LatLng(37.785, -122.437)),
+    const WeightedLatLng(LatLng(37.785, -122.435))
+  ];
+
+  List<WeightedLatLng> disabledPoints = <WeightedLatLng>[];
+
+  void _addPoint() {
+    if (disabledPoints.isEmpty) {
+      return;
+    }
+
+    final WeightedLatLng point = disabledPoints.first;
+    disabledPoints.removeAt(0);
+
+    setState(() => enabledPoints.add(point));
+  }
+
+  void _removePoint() {
+    if (enabledPoints.isEmpty) {
+      return;
+    }
+
+    final WeightedLatLng point = enabledPoints.first;
+    enabledPoints.removeAt(0);
+
+    setState(() => disabledPoints.add(point));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: <Widget>[
+        Center(
+          child: SizedBox(
+            width: 350.0,
+            height: 300.0,
+            child: GoogleMap(
+                initialCameraPosition: const CameraPosition(
+                  target: sanFrancisco,
+                  zoom: 13,
+                ),
+                heatmaps: <Heatmap>{
+                  Heatmap(
+                    heatmapId: const HeatmapId('test'),
+                    data: enabledPoints,
+                    gradient: HeatmapGradient(
+                      const <HeatmapGradientColor>[
+                        // Web needs a first color with 0 alpha
+                        if (kIsWeb)
+                          HeatmapGradientColor(
+                            Color.fromARGB(0, 0, 255, 255),
+                            0,
+                          ),
+                        HeatmapGradientColor(
+                          Color.fromARGB(255, 0, 255, 255),
+                          0.2,
+                        ),
+                        HeatmapGradientColor(
+                          Color.fromARGB(255, 0, 63, 255),
+                          0.4,
+                        ),
+                        HeatmapGradientColor(
+                          Color.fromARGB(255, 0, 0, 191),
+                          0.6,
+                        ),
+                        HeatmapGradientColor(
+                          Color.fromARGB(255, 63, 0, 91),
+                          0.8,
+                        ),
+                        HeatmapGradientColor(
+                          Color.fromARGB(255, 255, 0, 0),
+                          1,
+                        ),
+                      ],
+                    ),
+                    maxIntensity: 1,
+                    // Radius behaves differently on web and Android/iOS.
```

Comment:
Picking up from https://github.com/flutter/plugins/pull/5274#discussion_r960808709, this is the core sticking point with this PR from my perspective. The radius seems central to the heapmap, so if its behavior is unpredictable then it's going to be nearly impossible to use the feature correctly. And if we ship something that is wrong, we won't be able to fix it without a breaking change.

We need to understand why it's different (according to your tests, both OS to OS and version to version), whether the pixels are logical or device, etc. so that we have confidence in how we should handle it at the abstraction layer.

Reply:
@leighajarett Do we have any contacts on the Google Maps side who could help us understand the behavioral differences of heatmap pixel radius here?

Reply:
@amuramoto just following up on this, is someone on maps able to take a look?

Reply:
I'm still looking into how this work in iOS, but I think the difference between Android and web has to do with the meaning of 'radius' in each. 

For JS, radius actually sets the 'influence' of individual points, so the inner padding of each grid is equal to the value of `radius`, and will always be relative to the point(s) closest to the edge of the grid.

For Android and iOS, radius sets the width of the blur itself, so the intent is that it sets radius of the heatmap.

So, even though they are all in pixels, I _think_ the issue is that the reference point for applying the radius is different across platforms.

TBH - overall there are places where there is not exact parity cross-platform for Maps (a lot of this is because of the history of when/who/how these things were built).

Reply:
@amuramoto Are you aware of an easy way to automatically convert one value to the equivalent value each platform requires? Or should we create an object to hold a user-defined value for each platform?

Reply:
> Or should we create an object to hold a user-defined value for each platform?

I don't think we want to do a per-platform version, but we can break it down by conceptual meaning. If the web version is fundamentally different from mobile, then we can specify that separately (either a totally different parameter or an object with fields). But if iOS and Android are supposed to mean the same thing, we should figure out why they aren't and convert internally if at all possible.

Reply:
re: the differences between Android and iOS, I'm wondering if it is because of how max/min intensities are calculated. In the case of Android, an [assumed screen size of 1258px figures into the calculation](https://github.com/googlemaps/android-maps-utils/blob/main/library/src/main/java/com/google/maps/android/heatmaps/HeatmapTileProvider.java#L549), which [does not appear to be the case on iOS](https://github.com/googlemaps/google-maps-ios-utils/blob/main/src/Heatmap/GMUHeatmapTileLayer.m#L158). 

TBH I'm not really sure, but that's the one glaring difference I've been able to dig up between how the rendering is calculated between platforms.

@Rexios80 AFAIK, there isn't any set way to convert between them, but this screen size assumption on Android seems like a good starting point to look at.

Reply:
@stuartmorgan I can look into it again when I have time

Reply:
Note from triage: I also may be able to investigate this, but it will be at least a couple of weeks before I can do so.

Reply:
I tried playing with the `SCREEN_SIZE` value on Android using reflection and it seems to have no noticeable effect on the display

Reply:
Thinking about creating also an implementation auf heatmap as i otherwise need to switch to mapbox as they support this.

Reply:
First this isn't the place to discuss that. Second theres no reason you can't just use this fork. It's already working just not quite good enough to merge in yet.

Reply:
Since we still don't have any resolution on why these are different, but I don't want to prevent us from doing something better later if we are able to find a deterministic conversion/scaling in the future, here's my proposal to unblock this: we make the radius an object rather than a number, and we make its constructor private so that it has to be made via a factory. Something like:

```
class HeatmapRadius {
  HeatmapRadius._(this._pixels);
  /// Explanation of platform-specific nature here, with explanation of how to empirically determine values and example of how to set it per-platform.
  HeatmapRadius.fromPlatformSpecificValue(int radius) {...}
  final int _radius;
  int get pixels => _pixels;
```

We're still in the unfortunate decision of pushing the complexity of figuring this out to all of the plugin clients, but that's better than not having the feature at all. And that way, if we can determine later (perhaps in coopration with further investigation by engineers on the Google Maps team) that there is some deterministic conversion for at least some platforms, then we can later introduce a different factory method that does that, and potentially deprecate this one, all without breaking changes.

Reply:
Implementing this now

Reply:
Implemented the HeatmapRadius object